### PR TITLE
PLANET-7697 Add Posts/Actions List indicators only when needed

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7697](https://jira.greenpeace.org/browse/PLANET-7697)

This is for the carousel style

### Testing

Here is an [example page](https://www-dev.greenpeace.org/test-deimos/carousels/) with the different possibilities!

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [deimos](https://www-dev.greenpeace.org/test-deimos) | [admin](https://www-dev.greenpeace.org/test-deimos/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-deimos/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-deimos) | [composer-local.json](https://github.com/greenpeace/planet4-test-deimos/blob/main/composer-local.json)

:watch: 2024.12.12 15:08:22

File: assets/src/js/query_loop_carousel.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
 import {v4 as uuid} from 'uuid';
 
+const ARROW_DIRECTIONS = ['prev', 'next'];
+
+const removeArrows = layout => ARROW_DIRECTIONS.forEach(direction => {
+  const controlBtn = layout.querySelector(`.wp-block-buttons .carousel-control-${direction}`);
+  controlBtn?.parentNode.removeChild(controlBtn);
```
Comment (mardelnet) [inline assets/src/js/query_loop_carousel.js:7]: Should we check whether the `parentNode` also exists?

```suggestion
if (controlBtn && controlBtn.parentNode) {
  controlBtn.parentNode.removeChild(controlBtn);
}
```

File: assets/src/js/query_loop_carousel.js [hunk 2]
```diff
@@ -1,8 +1,16 @@
 import {v4 as uuid} from 'uuid';
 
+const ARROW_DIRECTIONS = ['prev', 'next'];
+
+const removeArrows = layout => ARROW_DIRECTIONS.forEach(direction => {
+  const controlBtn = layout.querySelector(`.wp-block-buttons .carousel-control-${direction}`);
```
Comment (mardelnet) [inline assets/src/js/query_loop_carousel.js:6]: Should we add an early return if `layout` is falsy?

File: assets/src/js/query_loop_carousel.js [hunk 3]
```diff
@@ -21,70 +29,76 @@ export const setupQueryLoopCarousel = () => {
...
+      if (posts.length > itemsPerSlide) {
+        indicators = document.createElement('ol');
+        indicators.classList.add('carousel-indicators');
+        carousel.append(indicators);
+
+        // Update controls
+        ARROW_DIRECTIONS.forEach(direction => {
+          const controlBtn = layout.querySelector(`.wp-block-buttons .carousel-control-${direction}`);
+          controlBtn.dataset.bsTarget = `#${uniqueId}`;
+          controlBtn.dataset.bsSlide = direction;
```
Comment (mardelnet) [inline assets/src/js/query_loop_carousel.js:47]: If `controlBtn` is not found, accessing `controlBtn.dataset` or `controlBtn.querySelector` will throw an error.

File: assets/src/js/query_loop_carousel.js [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
+
+const removeArrows = layout => ARROW_DIRECTIONS.forEach(direction => {
+  const controlBtn = layout.querySelector(`.wp-block-buttons .carousel-control-${direction}`);
+  controlBtn?.parentNode.removeChild(controlBtn);
+});
+
 export const setupQueryLoopCarousel = () => {
   for (const layout of document.querySelectorAll('[class*="is-custom-layout-"]')) {
     const list = layout.querySelector('.wp-block-post-template');
+    let indicators = null;
```
Comment (mardelnet) [inline assets/src/js/query_loop_carousel.js:13]: Should we add an early return if `layout` is falsy?

**Review by mardelnet — COMMENTED** [review]
Nice job, @mleray 
I have a couple of comments:

1. Can we add some JS Docs to this file?
2. I see there are a bunch of repeated values (`wp-block-buttons`, `carousel`, etc...). Should we create constants for them?
3. What about simplifying the code by replacing the nested conditions with guard classes?

File: assets/src/js/query_loop_carousel.js (hunk 2 above)
Comment (mleray) [reply assets/src/js/query_loop_carousel.js]: It should be okay, I've added a check for `layout` in the main function and we only use this one there

**Review by mardelnet — COMMENTED** [review]
LGTM @mleray 
Just out of curiosity before approving this, can't we add the arrows only when the conditions are met instead of always adding them and then removing them by calling the `removeArrows` function?

**Comment by mleray** [comment]
> LGTM @mleray Just out of curiosity before approving this, can't we add the arrows only when the conditions are met instead of always adding them and then removing them by calling the `removeArrows` function?

@mardelnet we could, my guess is that these elements are a bit more complicated with all the classes than the indicators, and that's why it was easier to define them directly in the block's inner blocks

Screenshot 2024-12-16 at 14 06 11

**Review by mardelnet — APPROVED** [review]

**Pull request merged**